src/test/java/org/jenkinsci/plugins/workflow/support/actions/LogActionImplTest.java

```diff
@@ -54,12 +54,12 @@
         WorkflowJob p = r.createProject(WorkflowJob.class);
         p.setDefinition(new CpsFlowDefinition("parallel a: {chatty('LBBL') {echo(/atom step in A with $remaining commands to go/)}}, b: {chatty('BL') {echo(/atom step in B with $remaining commands to go/)}}", true));
         WorkflowRun b = r.buildAndAssertSuccess(p);
-        r.assertLogContains("logging from LBBL with 3 commands to go", b);
-        r.assertLogContains("atom step in A with 2 commands to go", b);
-        r.assertLogContains("atom step in A with 1 commands to go", b);
-        r.assertLogContains("logging from LBBL with 0 commands to go", b);
-        r.assertLogContains("atom step in B with 1 commands to go", b);
-        r.assertLogContains("logging from BL with 0 commands to go", b);
+        r.waitForMessage("logging from LBBL with 3 commands to go", b);
+        r.waitForMessage("atom step in A with 2 commands to go", b);
+        r.waitForMessage("atom step in A with 1 commands to go", b);
+        r.waitForMessage("logging from LBBL with 0 commands to go", b);
+        r.waitForMessage("atom step in B with 1 commands to go", b);
+        r.waitForMessage("logging from BL with 0 commands to go", b);
```

Comment:
Replacing all occurrences is probably overkill. AFAIK the issue occurs when a message is printed right when the build is ending, such as the final error/stack trace in `WorkflowRun.finish`. So would suffice to patch just the cases observed to fail.